README.md

```diff
@@ -175,25 +168,47 @@ For information on how to send push notifcations through Klaviyo, please check o
 
 If you would like to track when a user opens a push notification then there is a little more code that you will need to add to your application.
 
-In your application delegate, under `application:didFinishLaunchingWithOptions:` add the following:
+In your application delegate, under `application(_:didReceiveRemoteNotification:fetchCompletionHandler:)` add the following:
 
 ```swift 
-    if let launch = launchOptions, let data = launch[UIApplicationLaunchOptionsKey.remoteNotification] as? [AnyHashable: Any] {
-        Klaviyo.sharedInstance.handlePush(userInfo: data as NSDictionary)
-    }
+    Klaviyo.sharedInstance.handlePush(userInfo: data as NSDictionary)
+    completionHandler(.noData)
 ```
 
-Under `application:didReceiveRemoteNotification:` add the following:
-``` 
-func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable: Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
-    
-    if application.applicationState == UIApplicationState.inactive || application.applicationState ==  UIApplicationState.background {
-        Klaviyo.sharedInstance.handlePush(userInfo: userInfo as NSDictionary)
+In addition please add the following code that extends your app delegate:
+
+```swift
+    extension AppDelegate: UNUserNotificationCenterDelegate {
+        func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
+            Klaviyo.sharedInstance.handlePush(userInfo: response.notification.request.content.userInfo as NSDictionary)
+            completionHandler()
+        }
     }
+    
 ```
 
 That is all you need to do to track opens. Now once your first push notifications have been sent and been opened, you should start to see `Opened Push` metrics within your Klaviyo dashboard.
 
+## [OPTIONAL] Foreground Push Handling
+
+The code below will enable push notification to show up when you app is running:
```

Comment:
nit: `will enable push notifications to show up when your app is running`

Just pluralizing "notification" and changing "you" for "your"

Reply:
Will update to:
```suggestion
The code below will enable push notifications to show up when you app is running:
```